src/app/portlets/dot-edit-content/components/dot-edit-content-view-as-toolbar/dot-edit-content-view-as-toolbar.component.spec.ts

```diff
@@ -0,0 +1,151 @@
+            ],
+            imports: [BrowserAnimationsModule],
+            providers: [
+                {
+                    provide: DotDevicesService,
+                    useClass: DotDevicesServiceMock
+                },
+                {
+                    provide: DotPersonasService,
+                    useClass: DotPersonasServiceMock
+                },
+                {
+                    provide: DotLanguagesService,
+                    useClass: DotLanguagesServiceMock
+                }
+            ]
+        });
+
+        fixture = DOTTestBed.createComponent(DotEditContentViewAsToolbarComponent);
+        component = fixture.componentInstance;
+        de = fixture.debugElement;
+        component.value = { language: mockDotLanguage };
+    });
+
+    it('should have Persona selector', () => {
+        const toolbarElement: DebugElement = de.query(By.css('dot-persona-selector'));
+        expect(toolbarElement).not.toBeNull();
+    });
+
+    it('should emit changes in Personas', () => {
+        const personaSelector: DebugElement = de.query(By.css('dot-persona-selector'));
+        spyOn(component, 'changePersonaHandler').and.callThrough();
+        spyOn(component.changeViewAs, 'emit');
+        personaSelector.componentInstance.selected.emit(mockDotPersona);
+        fixture.detectChanges();
+
+        expect(component.changePersonaHandler).toHaveBeenCalledWith(mockDotPersona);
+        expect(component.changeViewAs.emit).toHaveBeenCalledWith({language: mockDotLanguage, persona: mockDotPersona});
+    });
+
+    it('should have Device selector', () => {
+        const toolbarElement: DebugElement = de.query(By.css('dot-device-selector'));
+        expect(toolbarElement).not.toBeNull();
+    });
+
+    it('should emit changes in Device', () => {
+        const deviceSelector: DebugElement = de.query(By.css('dot-device-selector'));
+        spyOn(component, 'changeDeviceHandler').and.callThrough();
+        spyOn(component.changeViewAs, 'emit');
+        deviceSelector.componentInstance.selected.emit(mockDotDevice);
+        fixture.detectChanges();
+
+        expect(component.changeDeviceHandler).toHaveBeenCalledWith(mockDotDevice);
+        expect(component.changeViewAs.emit).toHaveBeenCalledWith({language: mockDotLanguage, device: mockDotDevice});
+    });
+
+    it('should have Language selector', () => {
+        const toolbarElement: DebugElement = de.query(By.css('dot-language-selector'));
+        expect(toolbarElement).not.toBeNull();
+    });
+
+    it('should emit changes in Language', () => {
+        const languageSelector: DebugElement = de.query(By.css('dot-language-selector'));
+        const testlanguage: DotLanguage = {
+            id: 2,
+            languageCode: 'es',
+            countryCode: 'es',
+            language: 'test',
+            country: 'test'
+        };
+        spyOn(component, 'changeLanguageHandler').and.callThrough();
+        spyOn(component.changeViewAs, 'emit');
```

Comment:
```
spyOn(component.changeViewAs, 'emit');
const languageSelector: DebugElement = de.query(By.css('dot-language-selector'));
const deviceSelector: DebugElement = de.query(By.css('dot-device-selector'));
const personaSelector: DebugElement = de.query(By.css('dot-persona-selector'));
```

This things can be easily move to the beforeEach because you are doing it in all the tests.